### PR TITLE
ConfigDocs: Document in/out_format

### LINKED ISSUE #176: Document in_format & out_format
Add them to ``config-v3-documentation.``

### PR DESCRIPTION
**Pull request by 5K1PY**
Fixes #176

### CODE REVIEW

File: config-v3-documentation [hunk 1]
```diff
@@ -55,6 +55,20 @@ data_subdir=tests/
...
+#           subject to normalization (newlines, encoding)
+#   binary  application/binary
+
+out_format=text
+# Format of output:
+#   text    text/plain in ASCII (default)
+#           subject to normalization (newlines, encoding)
+#   binary  application/binary
+#
+# TODO: Server mode
```
Comment (SvizelPritula) [inline config-v3-documentation:70]: I'm not sure I understand the point of this TODO. Will  the format or semantics of this option check anything when the server mode is implemented?

File: config-v3-documentation [hunk 2]
```diff
@@ -55,6 +55,20 @@ data_subdir=tests/
 in_gen=src/gen
 # Source code of generator (without suffix)
 
+in_format=text
+# Format of input:
+#   text    text/plain in ASCII (default)
+#           subject to normalization (newlines, encoding)
+#   binary  application/binary
```
Comment (SvizelPritula) [inline config-v3-documentation:62]: This is not a real MIME type. Did you mean `application/octet-stream`?

File: config-v3-documentation [hunk 3]
```diff
@@ -55,6 +55,20 @@ data_subdir=tests/
 in_gen=src/gen
 # Source code of generator (without suffix)
 
+in_format=text
+# Format of input:
+#   text    text/plain in ASCII (default)
```
Comment (SvizelPritula) [inline config-v3-documentation:60]: The `text/plain` MIME type requires that lines end with CRLF, but písek enforces LF.

**Review by SvizelPritula — CHANGES_REQUESTED** [review]

File: config-v3-documentation (hunk 3 above)
Comment (5K1PY) [reply config-v3-documentation:60]: Yup, that's why is there:
> subject to normalization (newlines, encoding)

File: config-v3-documentation (hunk 2 above)
Comment (5K1PY) [reply config-v3-documentation:62]: Right. But I would much prefer having something imediately clear...

File: config-v3-documentation (hunk 1 above)
Comment (5K1PY) [reply config-v3-documentation]: Improved the comment.

File: config-v3-documentation (hunk 2 above)
Comment (SvizelPritula) [reply config-v3-documentation:62]: If it's not supposed to be a MIME type, then I don't understand what you mean by "application/binary". Do you mean the input/output will be either an application or a sequence of bits?

File: config-v3-documentation (hunk 3 above)
Comment (SvizelPritula) [reply config-v3-documentation:60]: Sure, but písek doesn't normalize the files to `text/plain`. If we want our input files to be `text/plain` files, písek should replace all LFs by CRLFs before giving them as inputs to solutions or exporting them.

**Review by SvizelPritula — APPROVED** [review]
The new version makes way more sense, thanks!

**Pull request merged**